### ISSUE TITLE
Rule Change: only count function body

### ISSUE THREAD

**Issue by simplenotezy** [eslint/eslint#18786]
### What rule do you want to change?

max-lines-per-function

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
export const test = (
  httpServer: App,
  options: {
    id: Campaign['id'];
    input: UpdateCampaignInput;
    fields?: [keyof Campaign][];
    someOtherField: string;
    hello: string;
    a: string;
    asdasd: number;
    asd: boolean;
  },
): Test => {
  return graphQLRequest(httpServer, {
    query: '',
    variables: {
      id: options.id,
      input: options.input,
    },
  });
};
```

### What does the rule currently do for this code?

Arrow function has too many lines (21). Maximum allowed is 20.

### What will the rule do after it's changed?

It will only count the arrow functions body, i.e. count around 7 lines of code for this block:

```ts
return graphQLRequest(httpServer, {
  query: '',
  variables: {
    id: options.id,
    input: options.input,
  },
});
```

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

The max-lines-per-function call introduced here https://github.com/eslint/eslint/issues/9842 was great! But I think we need more options for taste; e.g. we use prettier, which sometimes will make the input arguments go on multiple lines, and that also counts as a new line in the `max-lines-per-function` call. I would expect the rule to only count the actual lines inside the body (everything within the brackets).

**Comment by nzakas** [comment]
Thanks for the input. As [stated in the README](https://github.com/eslint/eslint#stylistic-rule-updates), we actually aren't making any further updates to rules that enforce stylistic preferences. If you have a strong preference for such a change, you can always copy the existing rule into a custom rule and make it behave the way you want.
